adapter/internal/discovery/xds/common/utils_test.go

```diff
@@ -0,0 +1,75 @@
+/*
+ *  Copyright (c) 2022, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package common_test
```

Comment:
why do we want a different package name for tests?

Reply:
Changed it to common as now the method remains private. 
In golang tests, if we need to check private methods we use the same package name. But if we need to stick into public methods and test we use this pattern. 
example : https://github.com/envoyproxy/go-control-plane/blob/main/pkg/cache/v3/delta_test.go